### PR TITLE
Implement xdivy op

### PR DESCRIPTION
**Pull request by noim210**
implementation of xdivy op

### CODE REVIEW

File: test/test_math_ops.cpp [hunk 1]
```diff
@@ -2164,6 +2164,80 @@ TEST(MathOps, SquaredDifferenceBroadcasting) {
...
+TEST(MathOps,XdivyZeroX) {
+  Scope root = Scope::NewRootScope();
+  int dim1 = 2;
+  int dim2 = 2;
+
+  Tensor A(DT_FLOAT, TensorShape({dim1, dim2}));
+  Tensor B(DT_FLOAT, TensorShape({dim1, dim2}));
+
+  AssignInputValues(A, std::vector<float>{0.f, 1.f, 5.f, 3.21f ,8.1f });
+  AssignInputValues(B, std::vector<float>{2.0f, 1.2f, 4.2f, 8.9f});
```
Comment (sayantan-nervana) [inline test/test_math_ops.cpp]: Lets add tests where 
1. num is non zero positive and denom is 0
2. num is non zero negative and denom is 0

File: test/test_math_ops.cpp [hunk 2]
```diff
@@ -2164,6 +2164,80 @@ TEST(MathOps, SquaredDifferenceBroadcasting) {
...
+  vector<int> static_input_indexes = {};
+  auto R = ops::Xdivy(root, A, B);
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  std::vector<Output> sess_run_fetchoutputs = {R};
+  OpExecuter opexecuter(root, "Xdivy", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+  PrintTensor(A);
+  PrintTensor(B);
```
Comment (sayantan-nervana) [inline test/test_math_ops.cpp]: Please remove the prints

File: ngraph_bridge/ngraph_builder.cc [hunk 3]
```diff
@@ -4933,6 +4933,26 @@ static Status TranslateUnpackOp(const Node* op,
   return Status::OK();
 }
 
+static Status TranslateXdivyOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  
+  shared_ptr<ng::Node> ng_input1, ng_input2, ng_input3;
```
Comment (sayantan-nervana) [inline ngraph_bridge/ngraph_builder.cc]:
```suggestion
  shared_ptr<ng::Node> ng_input1, ng_input2;
```

File: ngraph_bridge/ngraph_builder.cc [hunk 4]
```diff
@@ -4933,6 +4933,26 @@ static Status TranslateUnpackOp(const Node* op,
...
+  TF_RETURN_IF_ERROR( GetInputNodes(ng_op_map, op, &ng_input1, &ng_input2));
+  std::tie(ng_input1, ng_input2) = Builder::PerformNgBroadcast(op->name(),ng_input1, ng_input2);
+  ng::Shape input1_shape = ng_input1->get_shape();
+  std::vector<std::string> const_values(ng::shape_size(input1_shape), "0");
+  auto const_zero= ConstructNgNode<ng::op::Constant>( op->name(), ng_input1->get_element_type(), input1_shape, const_values); 
+  auto ng_is_zero = ConstructNgNode<ng::op::Equal>(op->name(),ng_input1, const_zero);
+  auto ng_x_over_y = ConstructNgNode<ng::op::Divide>(op->name(), ng_input1, ng_input2);
+  auto ng_xdivy = ConstructNgNode<ng::op::Select>(op->name(),ng_is_zero, ng_input1, ng_x_over_y);
+  if(ng_xdivy!= ng_input1 && ng_xdivy!= ng_input2){
```
Comment (sayantan-nervana) [inline ngraph_bridge/ngraph_builder.cc]: We do not need this `if`
We need this if only in functions that accept ng nodes and return new ng nodes (and the function might be creating a small ng subgraph). Eg ngraph builders

File: test/test_math_ops.cpp [hunk 5]
```diff
@@ -2164,6 +2164,78 @@ TEST(MathOps, SquaredDifferenceBroadcasting) {
...
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  std::vector<Output> sess_run_fetchoutputs = {R};
+  OpExecuter opexecuter(root, "Xdivy", static_input_indexes, output_datatypes,
+                        sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+TEST(MathOps, XdivyZeroXZeroY) {
```
Comment (yimeisun123) [inline test/test_math_ops.cpp]: The 0/0 case is already included in the previous XdivyZeroX case as the last input pair. It is up to you to keep this test or remove it.

Comment (noim210) [reply test/test_math_ops.cpp]: Thanks I added as per your suggestion.  However, as pytest is not working i can't test it.

**Comment by adk9** [comment]
@googlebot I consent.

**Pull request merged**